easybuild/easyblocks/g/gcc.py

```diff
@@ -618,6 +619,52 @@ def post_install_step(self, *args, **kwargs):
             else:
                 raise EasyBuildError("Can't link '%s' to non-existing location %s", target, os.path.join(bindir, src))
 
+        # Rename include-fixed directory which includes system header files that were processed by fixincludes,
+        # since these may cause problems when upgrading to newer OS version.
+        # (see https://github.com/easybuilders/easybuild-easyconfigs/issues/10666)
+        glob_pattern = os.path.join(self.installdir, 'lib*', 'gcc', '*-linux-gnu', self.version, 'include-fixed')
+        matches = glob.glob(glob_pattern)
+        if matches:
+            if len(matches) == 1:
+                include_fixed_path = matches[0]
+
+                # limits.h and syslimits.h need to be copied to include/ first,
+                # these are strictly required (by /usr/include/limits.h for example)
+                include_path = os.path.join(os.path.dirname(include_fixed_path), 'include')
+                retained_header_files = ['limits.h', 'syslimits.h']
+                for fn in retained_header_files:
+                    from_path = os.path.join(include_fixed_path, fn)
+                    to_path = os.path.join(include_path, fn)
+                    if os.path.exists(from_path):
+                        if os.path.exists(to_path):
+                            raise EasyBuildError("%s already exists, not overwriting it with %s!", to_path, from_path)
+                        else:
+                            copy_file(from_path, to_path)
+                            self.log.info("%s copied to %s", from_path, to_path)
```

Comment:
Add the context to the log with regards? Something like "GCC include modifications:"?